### ISSUE TITLE
Should packages be explicit dependencies?

### ISSUE THREAD

**Issue by thlmenezes** [juliusmarminge/turbo-expo-next-starter#5]
Because it is a monorepo, I know that importing directly the packages will work because of the symlink inside `node_modules`; but shouldn't it be listed explicitly in the `packages.json`'s dependencies?

Example:
package api uses db, but db is not inside package.json

[packages/api/src/routers/context.ts](https://github.com/juliusmarminge/turbo-expo-next-starter/blob/ea734cdd5f5d62cf59ffce892eff3604a6d6d205/packages/api/src/routers/context.ts#L1)
```ts
import { PrismaClient } from "db";
```
[packages/api/package.json](https://github.com/juliusmarminge/turbo-expo-next-starter/blob/ea734cdd5f5d62cf59ffce892eff3604a6d6d205/packages/api/package.json#L8)
```json
"dependencies": {
  "@trpc/server": "^9.15.0",
  "zod": "^3.17.2"
}
```

**Comment by juliusmarminge** [comment]
> but shouldn't it be listed explicitly in the dependencies of the packages or apps?

Yes they should. This is one of the main things that I disliked about zart, all the dependencies just magically hoist together and it is definetely something I want to fix. Haven't had any time for this repo in a while now but if you want to hack on it feel free!

**Comment by thlmenezes** [comment]
Thanks for the reply and the awesome work with the repo, love the packages ideas (api and db packages specially); I've been hacking my own starter for my company so not much time either; but will come back later when finished and PR. Although it's way more opinionated, if you are curious it's on GitHub [here](https://github.com/tmtecnologia/turborepo-tmtecnologia).

I saw your work in the create-t3-app@next and got to this repo. Just starting on trpc and loving it.

**Comment by juliusmarminge** [comment]
> Thanks for the reply and the awesome work with the repo, love the packages ideas (api and db packages specially); I've been hacking my own starter for my company so not much time either; but will come back later when finished and PR. Although it's way more opinionated, if you are curious it's on GitHub [here](https://github.com/tmtecnologia/turborepo-tmtecnologia).
> 
> I saw your work in the create-t3-app@next and got to this repo. Just starting on trpc and loving it.

Please checkout [create-t3-turbo](https://github.com/juliusmarminge/create-t3-turbo) which has a much cleaner dependency tree imo.
